plugins/@grouparoo/mailchimp/src/lib/connect.ts

```diff
@@ -2,6 +2,12 @@ import { SimpleAppOptions } from "@grouparoo/core";
 import Mailchimp from "mailchimp-api-v3";
 
 export async function connect(appOptions: SimpleAppOptions) {
-  const client = new Mailchimp(appOptions.apiKey?.toString());
-  return client;
+  if (appOptions.apiKey) {
+    return new Mailchimp(appOptions.apiKey?.toString());
+  } else if (appOptions.oAuthToken) {
+    const [authToken, dc] = appOptions.oAuthToken.toString().split("::");
```

Comment:
Doing a bit of pre-processing on the token to extract the `dc` (datacenter).

Could also take in the oAuthToken plain and determine the `dc` here by doing a `meta` request, but would require adding something like axios. See my comments here: https://github.com/grouparoo/telemetry/pull/356#discussion_r783170760

Reply:
There's something a bit inflexible that's nagging at me using `::` as a delimiter.  Since we are building a deployed service with Telemetry, making changes there require more effort & coordination.

Is there any value to returning a (serialized) object from Telemetry rather than a :: separated string? If we did that, then we could add additional properties as we need without breaking older versions:

```ts
const {authToken, dc} = JSON.parse(appOptions.oAuthToken.toString()) // oAuthToken = {"authToken":"abc123","dc":"us-east"}
```

So, if one say we also needed to add "language" (us-en), for example, to the response from Telemetry, we could, and it would be backwards compatible.